### DISCUSSION TITLE
UpdateResult.Warnings not used anywhere?

### DISCUSSION THREAD

**Discussion by HenrikPoulsen** [renovatebot/renovate#15295] [Request Help]
Looking in the code it looks like the warnings accumulated in `lookupUpdates()` in the `UpdateResult.Warnings` is not used for anything.
Am I reading that correctly?
For example the `Failed to look up dependency ${depName}` warning is exceptionally important IMO. In my fork of renovate I changed the debug log for that so it always gets logged, and that has flushed out so many issues for us, where things have been misconfigured in various repos and lots of dependencies had not been updated for months.
Here's how I modified it for now:
```
        const warning: ValidationMessage = {
          topic: depName,
          message: `Failed to look up dependency ${depName}`,
        };
        logger.warn(
          {
            dependency: depName,
            packageFile,
            datasource,
            repository: config.repository,
            registryUrls: config.registryUrls,
          },
          warning.message
        );
```
so it includes more information than what was there originally, so you can see what the registry urls it was using etc was.

What is the intent with the Warnings? I would assume they are supposed to get printed somewhere at the end of the run, but currently doesn't seem to be used.

**Comment by rarkins** [comment]
I think this code may have been used originally but got dropped out at some point. It certainly would be useful to display them either in PRs or the Dependency Dashboard issue (collapsed, in case there's a lot)

**Reply by HenrikPoulsen** [reply]
Alright that makes sense.
Though for the dependency lookup error there would likely not be a PR I assume if it's not a flakiness thing.
Also in my org most repos have github issues turned off, because we use jira so that would likely cause these issues to still not be noticed (unless a PR would be created with these issues).
So I would argue having them logged as well would be useful, so that the information is visible somewhere in the case where there's no PR to write to and there's no dependency dashboard

**Reply by rarkins** [reply]
I think it could be beneficial to log in all PRs if there are warnings during a run not directly related to the current branch, although perhaps this could be opt-in. And yes I'm not against logging them.
